### ISSUE TITLE
React has detected a change in the order of Hooks called by SpaceInternal

### ISSUE THREAD

**Issue by thekapadia** [aeagle/react-spaces#37]
@aeagle I get this error on the console when I change the *zIndex* programmatically.

```
Warning: React has detected a change in the order of Hooks called by SpaceInternal. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://fb.me/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useRef                     useRef
2. useState                   useState
3. useRef                     useRef
4. useRef                     useRef
5. useEffect                  useEffect
6. useEffect                  useEffect
7. useEffect                  useEffect
8. useContext                 useContext
9. undefined                  useContext
```

**Comment by aeagle** [comment]
@thekapadia I believe I have resolved the issue in version 0.1.18. I was using a hook within a conditional expression.

**Comment by thekapadia** [comment]
@aeagle Yes, this error does not occur in 0.1.18. Thanks!
